cli/cmd/sync.go

```diff
@@ -257,3 +261,31 @@ func syncConnectionV1(ctx context.Context, cqDir string, sourceClient *clients.S
 func isUnknownConcurrencyFieldError(err error) bool {
 	return strings.Contains(err.Error(), unknownFieldErrorPrefix+`"table_concurrency"`) || strings.Contains(err.Error(), unknownFieldErrorPrefix+`"resource_concurrency"`)
 }
+
+func checkPluginsCompatibility(ctx context.Context, sourcePluginName string, sourcePluginVersion string, expectedProtocolVersion uint64, destClients destinationClients) error {
+	message := ""
+
+	for _, destClient := range destClients {
+		destPluginName, err := destClient.Name(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get destination name: %w", err)
+		}
+		destPluginVersion, err := destClient.Version(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get destination version: %w", err)
+		}
+
+		destProtocolVersion, err := destClient.GetProtocolVersion(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get protocol version from destination %q: %w", destPluginName+"@"+destPluginVersion, err)
+		}
+		if destProtocolVersion != expectedProtocolVersion {
```

Comment:
Initially I thought about trying to be more specific (e.g. telling the user to update only the destination or only the source), but that probably has a ton of edge cases. For example what if we have protocol version 3, then we need to start to check ranges 🙃